test/tools/Modules/PSSysLog/PSSysLog.psm1

```diff
@@ -0,0 +1,906 @@
+    {
+        throw "This command requires Linux"
+    }
+}
+
+function Start-NativeExecution
+{
+    param
+    (
+        [Parameter(Mandatory)]
+        [ValidateNotNull()]
+        [ScriptBlock] $command
+    )
+    $saveErrorAction = $ErrorActionPreference
+    $ErrorActionPreference = 'Continue'
+
+    try
+    {
+        & $command
+        if ($LASTEXITCODE -ne 0)
+        {
+            throw "Execution of {$command} failed with exit code $LASTEXITCODE"
+        }
+    }
+    finally
+    {
+        $ErrorActionPreference = $saveErrorAction
+    }
+}
+
+#endregion Utilities
+
+#region SysLog support
+
+# Defines the array indices when calling
+# String.Split on a SysLog log entry
+enum SysLogIds
+{
+    Month = 0;
+    Day = 1;
+    Time = 2;
+    Hostname = 3;
+    Id = 4;
+    CommitId = 5;
+    EventId = 6;
+    Message = 7;
+}
+
+# Defines the array indices when calling
+# String.Split on an OsLog log entry
+enum OsLogIds
+{
+    Date = 0;
+    Time = 1;
+    Thread = 2;
+    Type = 3;
+    Activity = 4;
+    Pid = 5;
+    ProcessName = 6;
+    Module = 7;
+    Id = 8;
+    CommitId = 9;
+    EventId = 10;
+    Message = 11;
+}
+
+class PSLogItem
+{
+    [string] $LogId = [string]::Empty
+    [DateTime] $Timestamp = [DateTime]::Now
+    [string] $Hostname = [string]::Empty
+    [int] $ProcessId = 0
+    [int] $ThreadId = 0
+    [int] $Channel = 0
+    [string] $CommitId = [string]::Empty
+    [string] $EventId = [string]::Empty
+    [string] $Message = [string]::Empty
+    [int] $Count = 1
+
+    hidden static $monthNames = @('Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun','Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec')
+
+    hidden static [int] GetMonth([string] $value)
+    {
+        Set-StrictMode -Version Latest
+        for ($x = 0; $x -lt [PSLogItem]::monthNames.Count; $x++)
+        {
+            [string] $monthName = [PSLogItem]::monthNames[$x]
+            if ($value.StartsWith($monthName, [StringComparison]::InvariantCultureIgnoreCase))
+            {
+                return $x + 1
+            }
+        }
+        return 1
+    }
+
+    static [PSLogItem] ConvertSysLog([string] $content, [string] $id, [Nullable[DateTime]] $after)
+    {
+        Set-StrictMode -Version Latest
+        <#
+        MMM dd HH:MM:SS machinename id[PID]: (commitid:TID:CHANNEL) [EventName] Message
+        Expecting split to return
+        0: Month (abreviated)
+        1: DD
+        2: HH:MM:SS
+        3: hostname
+        4: nameid[processid]:
+        5: (commitid:treadid:channel)
+        6: [EventId]
+        7: Message Text
+
+        NOTE: In some cases, syslog will detect the same message being logged multiple times.
+        In those cases, a single message is logged in the following format
+
+        MMM dd HH:MM:SS machinename id[PID]: message repeated NNN times: [(commitid:TID:CHANNEL) [EventName] Message]
+        #>
+        [string[]] $parts = $content.Split(' ', 8, [System.StringSplitOptions]::RemoveEmptyEntries)
```

Comment:
Add comments explaining what you are doing... `# Splitting content into 8 fields delimited by a space`

Reply:
Will add an explicit intent comment.

Reply:
Fixed

Reply:
closed